external/emulation/emulation/models.py

```diff
@@ -32,6 +32,18 @@ def __call__(self, state: FortranState):
         return model_outputs
 
 
+def transform_model(
+    model: Callable[[FortranState], FortranState], transform: Any
+) -> Callable[[FortranState], FortranState]:
+    def combined(x: FortranState) -> FortranState:
+        x_transformed = _predict(transform.forward, x)
+        x_transformed.update(model(x_transformed))
```

Comment:
Yah. the type hints might clarify things a bit. I agree we could pull the transpose operation to a higher level so that we can work in the more predominant convention.